frontend/js/pages/DetailBattle.js

```diff
@@ -23,36 +26,21 @@ const StyledVersus = styled.span`
 `;
 
 class DetailBattle extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      battle: {
-        winner: '',
-        id: 0,
-        creator_team: {
-          trainer: '',
-          team: [],
-        },
-        opponent_team: {
-          trainer: '',
-          team: [],
-        },
-      },
-    };
-  }
-
   componentDidMount() {
+    const { getBattle } = this.props;
     // TODO: Will use pathname from props when I implement react-router-dom
     const { pathname } = window.location;
     const id = pathname.slice(pathname.lastIndexOf('/') + 1);
     axios.get(`/api/battle/${id}`).then((response) => {
-      this.setState({ battle: response.data });
-      return response.data;
+      return getBattle(response.data);
     });
   }
 
   render() {
-    const { battle } = this.state;
+    const { battle, isLoading } = this.props;
+    if (isLoading) {
+      return <>Loading</>;
+    }
```

Comment:
When we have a condition with just a return in its scope, we can write it like this:

```suggestion
    if (isLoading) return <>Loading</>;
```

But feel free to use it as you prefer, it's just a note :)